Userland/Libraries/LibWeb/Painting/BackgroundPainting.cpp

```diff
@@ -93,6 +133,16 @@ void paint_background(PaintContext& context, Layout::NodeWithStyleAndBoxModelMet
         painter.add_clip_rect(clip_rect);
         ScopedCornerRadiusClip corner_clip { painter, clip_rect, clip_box.radii };
 
+        if (layer.background_image->is_linear_gradient()) {
+            // FIXME: Paint non-orthogonal gradients, gradients with > 2 color stops, gradients with custom stop positions ...
+            auto& linear_gradient = layer.background_image->as_linear_gradient();
+            auto gfx_gradient = linear_gradient_to_gfx_gradient(linear_gradient, border_box.rect);
+            if (gfx_gradient.has_value()) {
+                painter.fill_rect_with_gradient(gfx_gradient->orientation, border_box.rect.to_rounded<int>(), gfx_gradient->color_a, gfx_gradient->color_b);
+            }
+            continue;
+        }
```

Comment:
Gradient backgrounds follow the same positioning, clipping and sizing rules as regular images, so this needs to happen after all those calculations. Which is quite a bit of extra work, sorry!

(As always, you can just put a `// FIXME:` on this if you don't want to deal with it right away.)

Reply:
The clipping already works, the rest is a +1 :bison: :)